From 69a5464e01a80fb0f0f76232381a9fd45f33a6e6 Mon Sep 17 00:00:00 2001
From: Fedora systemd team <systemd-maint@redhat.com>
Date: Mon, 26 Jan 2015 08:37:36 +0000
Subject: [PATCH] CA-159050: journal: Disable kmsg device lookups

Prevent deadlock when a VIF comes online where journald tries to capture
the uevent for the VIF that is mentioned in the kmsg and xenstored tries
to log to the journal, but reading the uevent requires xenstored.
---
 src/journal/journald-kmsg.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/journal/journald-kmsg.c b/src/journal/journald-kmsg.c
index 9895808..890c063 100644
--- a/src/journal/journald-kmsg.c
+++ b/src/journal/journald-kmsg.c
@@ -216,6 +216,7 @@ static void dev_kmsg_record(Server *s, char *p, size_t l) {
                 k = e + 1;
         }
 
+#if 0
         if (kernel_device) {
                 struct udev_device *ud;
 
@@ -265,6 +266,7 @@ static void dev_kmsg_record(Server *s, char *p, size_t l) {
                         udev_device_unref(ud);
                 }
         }
+#endif
 
         if (asprintf(&source_time, "_SOURCE_MONOTONIC_TIMESTAMP=%llu", usec) >= 0)
                 IOVEC_SET_STRING(iovec[n++], source_time);
-- 
1.9.3

